sources/packages/backend/workflow/test/2021-2022/assessment-gateway/assessment-gateway.originalAssessment.e2e-spec.ts

```diff
@@ -0,0 +1,87 @@
+import { ASSESSMENT_ID } from "@sims/services/workflow/variables/assessment-gateway";
+import {
+  ApplicationExceptionStatus,
+  AssessmentTriggerType,
+} from "@sims/sims-db";
+import { AssessmentConsolidatedData } from "../../models";
+import { ZBClient } from "zeebe-node";
+import {
+  createFakeConsolidatedFulltimeData,
+  ZeebeMockedClient,
+  PROCESS_INSTANCE_CREATE_TIMEOUT,
+  WorkflowServiceTasks,
+  createFakeSingleIndependentStudentData,
+} from "../../test-utils";
+import { PROGRAM_YEAR } from "../constants/program-year.constants";
+
+describe(`E2E Test Workflow assessment gateway on original assessment for ${PROGRAM_YEAR}`, () => {
+  let zeebeClientProvider: ZBClient;
+  beforeAll(async () => {
+    zeebeClientProvider = ZeebeMockedClient.getMockedZeebeInstance();
+  });
+
+  it("Should follow the expected workflow path when student is single and independent without application exception and PIR.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedFulltimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.assessmentTriggerType =
+      AssessmentTriggerType.OriginalAssessment;
+
+    const loadConsolidatedDataResult: Partial<AssessmentConsolidatedData> =
+      createFakeSingleIndependentStudentData();
+    // Application with no exception.
+    loadConsolidatedDataResult.applicationExceptionStatus =
+      ApplicationExceptionStatus.Approved;
+    // Application with PIR not required.
+    loadConsolidatedDataResult.studentDataSelectedOffering = 1;
+
+    // Act/Assert
+    const assessmentGatewayResponse =
+      await zeebeClientProvider.createProcessInstanceWithResult({
+        bpmnProcessId: "assessment-gateway",
+        variables: {
+          ...assessmentConsolidatedData,
+          [ASSESSMENT_ID]: 1,
+          // Data that will be returned by the worker that subscribe to load assessment data service task.
+          [`${WorkflowServiceTasks.LoadAssessmentConsolidatedData}-result`]:
+            loadConsolidatedDataResult,
+        },
+        requestTimeout: PROCESS_INSTANCE_CREATE_TIMEOUT,
+      });
+    // Workflow instance expected to pass through associate workflow instance service task.
+    expect(
+      assessmentGatewayResponse.variables[
+        WorkflowServiceTasks.AssociateWorkflowInstance
```

Comment:
This would be the moment where we should use the "service task id". Are we planning to do it in the upcoming PR?

Reply:
It is using the service task id

Reply:
Yes, I was misled by the `AssociateWorkflowInstance` and did not realize the `WorkflowServiceTasks` enum, my bad.